src/hooks/usePagination.js

```diff
@@ -1,14 +1,31 @@
 export const DOTS = "...";
 
-function usePagination() {
-  /*
-    Rewrite the logic here to map out the pagination to be displayed
+function usePagination({ currentPage, pageSize, totalCount }) {
```

Comment:
The value returned by this hook can be memoized/cached to avoid unnecessarily running the logic every time the hook is called. The useMemo() hook seems suitable.